isis/src/base/apps/spkwriter/SpkSegment.cpp

```diff
@@ -168,6 +169,42 @@ void SpkSegment::import(Cube &cube) {
     setStartTime(m_epochs[0]);
     setEndTime(m_epochs[size(m_epochs)-1]);
 
+    Pvl *label = cube.label();
+    QString labStartTime = getKeyValue(*label, "StartTime");
+    QString labEndTime;
+    QString value = getKeyValue(*label, "StopTime");
+    if (!value.isEmpty()) {
+      labEndTime = value;
+    }
+    else {
+      labEndTime = labStartTime;
+    }
+    iTime etLabStart(labStartTime);
```

Comment:
Using label start/stopTime could cause problems in some missions where the start/stop Times are not accurate. It would be more accurate to use Start/Stop ClockCounts.